executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -64,18 +67,44 @@ pub fn split_out_machines<'a, T: FieldElement>(
     let mut base_identities = identities.clone();
     let mut extracted_prover_functions = HashSet::new();
     let mut id_counter = 0;
-    for id in &identities {
-        // Extract all witness columns in the RHS of the lookup.
-        let lookup_witnesses = match id {
-            Identity::Lookup(LookupIdentity { right, .. })
-            | Identity::PhantomLookup(PhantomLookupIdentity { right, .. })
-            | Identity::Permutation(PermutationIdentity { right, .. })
-            | Identity::PhantomPermutation(PhantomPermutationIdentity { right, .. }) => {
-                &refs_in_selected_expressions(right) & (&remaining_witnesses)
+
+    let all_connections = identities
+        .iter()
+        .filter_map(|i| Connection::try_from(*i).ok())
+        .collect::<Vec<_>>();
+
+    let mut fixed_lookup_connections = BTreeMap::new();
+
+    for connection in &all_connections {
+        // If the RHS only consists of fixed columns, record the connection and continue.
+        if connection.is_lookup()
```

Comment:
Maybe extract the condition into a function?

Reply:
Right. I added a `FixedLookup::is_responsible`. In the near future, it should just have a `try_new()` like any other machine...